src/iris/api.py

```diff
@@ -3782,42 +3782,58 @@ def on_post(self, req, resp):
             raise HTTPBadRequest('Invalid keys in payload')
 
         affected = False
-        with db.guarded_session() as session:
-            affected = session.execute(
-                '''UPDATE `twilio_delivery_status`
-                   SET `status` = :status
-                   WHERE `twilio_sid` = :sid''',
-                {'sid': sid, 'status': status}).rowcount
-            session.commit()
+        connection = db.engine.raw_connection()
+        cursor = connection.cursor()
+        try:
+            max_retries = 3
+            for i in xrange(max_retries):
+                try:
+                    affected = cursor.execute(
+                        '''UPDATE `twilio_delivery_status`
+                           SET `status` = %(status)s
+                           WHERE `twilio_sid` = %(sid)s''',
+                        {'sid': sid, 'status': status})
+                    connection.commit()
+                    break
+                except Exception:
+                    logger.exception('Failed running Twilio status query. (Try %s/%s)', i + 1, max_retries)
```

Comment:
When this fails do we need to create a new cursor object? I think I remember somewhere else in the codebase needing to create a new connection.cursor() whenever the old one fails and dies.

Reply:
It doesn't look like it. We use this pattern in iris-sender a few times, and we're never re-creating the cursor there.